Assets/Scripts/Interactables/Dialogue/NPCDialogue/NPCInteractable.cs

```diff
@@ -1,62 +1,114 @@
-﻿using System.Collections;
-using System.Collections.Generic;
-using UnityEngine;
-
-[System.Serializable]
-public class MissionNPC
-{
-    public GameObject missionPrefab;
-
-    // Type of conversations an NPC can have based on if it is before, during or after a
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+[System.Serializable]
+public class NPCMissionConvos
+{
+    public GameObject missionPrefab;
+
+    // Type of conversations an NPC can have based on if it is before, during or after a
     public Conversation beforeConvo;
     public Conversation duringConvo;
     public Conversation afterConvo;
-}
-
-public class NPCInteractable : DialogueInteractable
-{
-    public List<Conversation> defaultConvos;
-
-    public List<MissionNPC> missionsOffered;
-    private MissionNPC currentMissionNPC;                            // Current Mission given by this NPC (should only be 1 per NPC)
-    private AMission startedMission;                                 // started mission needed to end mission
-
-
-    public override void OnInteract()
+}
+
+public class NPCInteractable : DialogueInteractable
+{
+    public List<Conversation> defaultConvos;
+
+    public List<NPCMissionConvos> missionsOffered;
+    private NPCMissionConvos currentMissionConvos = null;            // Current Mission given by this NPC (should only be 1 per NPC)
+    private AMission startedMission;                                 // started mission needed to end mission
+
+
+    protected NavMeshAgent agent;
+    protected GameObject targetObject;
+
+    protected bool isFollowing = false;
+    private Vector3 originPosition;
+
+
+    protected override void Start()
     {
-        if (!isConversing)
+        base.Start();
+        agent = Utils.GetRequiredComponent<NavMeshAgent>(this);
+        originPosition = gameObject.transform.position;
+        LoadConversation();
+    }
+
+    protected override void Update()
+    {
+        if(IsWithinBoundaryRadius(GameManager.Instance.GetPlayerTransform()))
         {
-            // Load the conversation of the NPC based on mission progress
-            if (missionsOffered.Count == 0)
+            LoadConversation();
+
+            // Autoplay
+            if(conversation.isAutoplayed && !autoPlaying)
             {
-                // Return a random default convo (Temporary until better system is set for default convos)
-                conversation = defaultConvos[Random.Range(0, defaultConvos.Count)];
+                OnInteract();
             }
-            // NPC has mission to offer
+            // Enter collider to interact
             else
+            {
+                base.Update();
+            }
+        }
+
+        if(isFollowing)
+        {
+            FollowTarget();
+
+            if(!IsWithinBoundaryRadius(GameManager.Instance.GetPlayerTransform()))
+            {
+                StopFollow();
+            }
+        }
```

Comment:
Maybe we should just add the logic of `StopFollow` in an else to the first `IsWithinBoundaryRadius` check? Distance checks are always expensive, especially when they add up with multiple NPCs, so maybe we can keep it to one call here.